sstables/index_reader.hh

```diff
@@ -760,18 +762,29 @@ private:
         return options;
     }
 
+    std::unique_ptr<partition_index_cache> make_local_index_cache(use_caching caching) const {
+        switch (caching) {
+        case use_caching::global:
+            return nullptr;
+        case use_caching::local:
+            return std::make_unique<partition_index_cache>(_sstable->manager().get_cache_tracker().get_lru(),
+                                                           _sstable->manager().get_cache_tracker().region());
+        case use_caching::none:
+            // No LRU attached, keeps only used entries.
+            return std::make_unique<partition_index_cache>(_sstable->manager().get_cache_tracker().region());
+        }
+        __builtin_unreachable();
+    }
 public:
     index_reader(shared_sstable sst, reader_permit permit,
                  tracing::trace_state_ptr trace_state = {},
-                 use_caching caching = use_caching::yes,
+                 use_caching caching = use_caching::global,
                  bool single_partition_read = false)
         : _sstable(std::move(sst))
         , _permit(std::move(permit))
         , _trace_state(std::move(trace_state))
-        , _local_index_cache(caching ? nullptr
-            : std::make_unique<partition_index_cache>(_sstable->manager().get_cache_tracker().get_lru(),
-                                                      _sstable->manager().get_cache_tracker().region()))
-        , _index_cache(caching ? *_sstable->_index_cache : *_local_index_cache)
+        , _local_index_cache(make_local_index_cache(caching))
+        , _index_cache(caching == use_caching::global ? *_sstable->_index_cache : *_local_index_cache)
```

Comment:
instead of comparing `caching` with `use_caching::global`, how about checking if `_local_index_cache` is nullptr? more intuitive this way, and less repeating.